src/python/pants/backend/javascript/dependency_inference/rules.py

```diff
@@ -110,8 +129,10 @@ async def infer_js_source_dependencies(
     owning_targets = await Get(Targets, Addresses(owners))
 
     non_path_string_bases = FrozenOrderedSet(
-        os.path.basename(non_path_string) for non_path_string in import_strings - path_strings
+        non_path_string.split(os.path.sep, 1)[0]
```

Comment:
Nit: For `split(.., 1)`, `partition` can be a little bit cleaner: https://docs.python.org/3/library/stdtypes.html#str.partition